src/app/static/src/app/store/baseline/mutations.ts

```diff
@@ -49,7 +54,7 @@ export const mutations: MutationTree<BaselineState> & BaselineMutations = {
     },
 
     PopulationUploadError(state: BaselineState, action: PayloadWithType<string>) {
-        state.populationError = action.payload;
+        state.populationError = Object.freeze(action.payload);
```

Comment:
Is it the case that we actually want to freeze anything coming back from the API (it's not going to change until we re-request). Could the apiService return objects pre-frozen so that all the mutations don't have to freeze the payloads? 

Would we still need to remember to freeze parts of object (like filters.regions.children..)? Freeze just freezes the top layer of the object? I guess we could implement a recursive freeze?

Reply:
Good idea; yes it would have to be recursive. Will add.